File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJImplementationCategoryDeclarationsIndex.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.stubs.StubIndexKey;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJImplementationDeclaration;
import org.jetbrains.annotations.NotNull;

public class ObjJImplementationCategoryDeclarationsIndex extends ObjJStringStubIndexBase<ObjJImplementationDeclaration> {

    private static final ObjJImplementationCategoryDeclarationsIndex INSTANCE = new ObjJImplementationCategoryDeclarationsIndex();
    private static final StubIndexKey<String, ObjJImplementationDeclaration> KEY = IndexKeyUtil.createIndexKey(ObjJImplementationCategoryDeclarationsIndex.class);
    private static final int VERSION = 0;

    private ObjJImplementationCategoryDeclarationsIndex(){}

    public static ObjJImplementationCategoryDeclarationsIndex getInstance() {
        return INSTANCE;
    }

    @Override
    public int getVersion() {
        return super.getVersion() + VERSION;
    }

    @NotNull
    @Override
    protected Class<ObjJImplementationDeclaration> getIndexedElementClass() {
        return ObjJImplementationDeclaration.class;
    }

    @NotNull
    @Override
    public StubIndexKey<String, ObjJImplementationDeclaration> getKey() {
        return KEY;
    }

}

File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJImplementationCategoryDeclarationsIndex.kt
package org.cappuccino_project.ide.intellij.plugin.indices

import com.intellij.psi.stubs.StubIndexKey
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJImplementationDeclaration

class ObjJImplementationCategoryDeclarationsIndex private constructor() : ObjJStringStubIndexBase<ObjJImplementationDeclaration>() {

    protected override val indexedElementClass: Class<ObjJImplementationDeclaration>
        get() = ObjJImplementationDeclaration::class.java

    override fun getVersion(): Int {
        return super.getVersion() + VERSION
    }

    override fun getKey(): StubIndexKey<String, ObjJImplementationDeclaration> {
        return KEY
    }

    companion object {

        val instance = ObjJImplementationCategoryDeclarationsIndex()
        private val KEY = IndexKeyUtil.createIndexKey(ObjJImplementationCategoryDeclarationsIndex::class.java)
        private val VERSION = 1
    }

}
